src/main/java/io/github/noeppi_noeppi/mods/bongo/EventListener.java

```diff
@@ -193,35 +191,28 @@ public void resourcesReloaded(OnDatapackSyncEvent event) {
 
     @SubscribeEvent
     public void damage(LivingHurtEvent event) {
-        if (!event.getEntityLiving().getCommandSenderWorld().isClientSide && event.getEntityLiving() instanceof Player player && !event.getSource().isBypassInvul()) {
-            Bongo bongo = Bongo.get(player.getCommandSenderWorld());
-            Team team = bongo.getTeam(player);
-            if (bongo.running() && team != null) {
-                if (event.getSource().getEntity() instanceof Player source) {
-                    if (!bongo.getSettings().pvp) {
-                        event.setCanceled(true);
-                    } else if (team.hasPlayer(source)) {
-                        if (!bongo.getSettings().friendlyFire) {
-                            event.setCanceled(true);
-                        }
-                    }
-                } else if (bongo.getSettings().invulnerable) {
-                    event.setCanceled(true);
-                }
-            }
-        }
+        handleEvent(event);
     }
 
     @SubscribeEvent
     public void attack(LivingAttackEvent event) {
-        if (!event.getEntityLiving().getCommandSenderWorld().isClientSide && event.getEntityLiving() instanceof Player player && !event.getSource().isBypassInvul()) {
+        handleEvent(event);
+    }
+
+    private void handleEvent(LivingEvent event) {
+        DamageSource source = fromEvent(event);
```

Comment:
Add the damage source as a parameter here instead of extracting it from the event later.
Also the method needs a better name, maybe name it `commonDamage`